Make all object types have the same size limits

mirrorwish opened the pull request:
The size limits of objects currently vary depending on their type. This makes it harder to create new versions of old object types as for instance a `pubkey` longer than 440 bytes will not be relayed. So I think the only limit should be the limit of 2**18 bytes.

Commit:
Same size limits for all objects

Commit:
Merge remote-tracking branch 'upstream/master'

PeterSurda commented:
I can review the code but I need to discuss with other people if the limit removal is a good idea.

mailchuck committed:
Newly arrives messages sorting

There could be cases where newly arrives message is not added correctly.
It won't necessarily go to the top, that depends on the sort.

mailchuck committed:
Thread names for IPv6

Thread names for IPv6 contained ":". This caused problems in log
parsers.

mailchuck committed:
private IP range checks

When advertising nodes and when establishing connections, private IP
range checks were not done. This could cause private IPs to be
advertised across the network. Also, some of the checks weren't
IPv6-aware.
Fixes Bitmessage#768

mailchuck committed:
Cleanup lockfile on exit

singleton.py design was broken.
Fixed Bitmessage#775

mailchuck committed:
Unused variable

Commit:
Merge branch 'master' of https://github.com/mirrorwish/PyBitmessage

ghost commented:
This should not be changed unless ACKs are simultaneously limited to only be of object type `msg`. But it might make sense to document the current behavior.

For reference these are the current limits as implemented by PyBitmessage:

```
getpubkey (0): 42 <= size <= 200
pubkey (1): 146 <= size <= 440
msg (2): size <= 2**18
broadcast (3): 180 <= size <= 2**18 and version >= 2
anything else: size <= 2**18
```
